File: 프로그래머스/lv1/12921. 소수 찾기/소수 찾기.java
class Solution {
    public int solution(int n) {

        int[] arr = new int[n + 1];
        int answer = 0;

        for (int i = 2; i < Math.sqrt(n) + 1; i++) {
            if (arr[i] == 0) {
                int j = 2;
                while (j * i <= n) {
                    arr[i * j] = 1;
                    j++;
                }
            }
        }

        for (int i = 2; i < n + 1; i++) {
            if(arr[i] == 0) {
                answer++;
            }
        }

        return answer;
    }
}

File: 프로그래머스/lv2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations
import math

def solution(numbers):

    arr = []
    for i in range(len(numbers)):
        arr.append(numbers[i])
    
    result = set()
    
    for length in range(1,len(arr)+1):
        data = list(permutations(arr,length))
        # print(data)
        for i in range(len(data)):
            number = int("".join(data[i]))
            # print(number)
            if number != 0 and number != 1:
                result.add(number)
            
    result = list(result)
    
    for num in range(len(result)):
        for i in range(2,result[num]):
            if result[num] % i == 0:
                result[num] = 0
                break

    answer = 0
    for num in result:
        if num != 0:
            answer += 1
            
    return answer